src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -919,5 +919,41 @@ void CodegenLLVMVisitor::visit_while_statement(const ast::WhileStatement& node)
     builder.SetInsertPoint(exit);
 }
 
+void CodegenLLVMVisitor::wrap_kernel_function(const std::string& kernel_name) {
+    // Get the kernel function and the instance struct type.
+    auto kernel = module->getFunction(kernel_name);
+    if (!kernel)
+        throw std::runtime_error("Kernel " + kernel_name + " is not found!");
+
+    if (kernel->getNumOperands() != 0)
```

Comment:
I am wondering how this is working. Does the `getNumOperands()` give the number of arguments of `llvn::Function`? Is yes, shouldn't this be one (the `InstanceStruct*`)?

Reply:
That's a mistake! Thanks! `getNumOperands()` comes from `llvm::Value` class, so in case of functions this turns out to be true. What I meant here was the number of arguments (basically `args()`) which indeed should be 1.